streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBStore.java

```diff
@@ -283,7 +283,11 @@ public synchronized V putIfAbsent(K key, V value) {
     private void restoreAllInternal(Collection<KeyValue<byte[], byte[]>> records) {
         try (WriteBatch batch = new WriteBatch()) {
             for (KeyValue<byte[], byte[]> record : records) {
-                batch.put(record.key, record.value);
+                if (record.value == null) {
+                    batch.remove(record.key);
```

Comment:
What's the semantics of `WriteBatch#remove`, does that just remove the key from this batch or is it adding a delete operation to the underlying store? If it is for the former case only then we cannot rely on this function since records with the same key may not fall in the same batch during restoration.